llm/turbine_llm/models/llama.py

```diff
@@ -4,30 +4,72 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+from typing import Optional
+
+import math
+
 import torch
 import torch.nn as nn
+import torch.nn.functional as F
 
 from ..config.llm_configs import LlamaHParams
 from ..data import Theta
-from ..layers.attention import (
-    LlamaAttentionBlock,
-    RotaryEmbeddingLayer,
-)
-
-from ..layers.core import (
+from ..layers import (
     LinearLayer,
     RMSNormLayer,
     ThetaLayer,
     TokenEmbedding,
+    RotaryEmbeddingLayer,
+    PagedKVCache,
 )
 
+from .base import BaseCausalLMModel
+
+__all__ = [
+    "PagedLlamaModelV1",
+]
+
+
+################################################################################
+# Models
+################################################################################
+
+
+class PagedLlamaModelV1(BaseCausalLMModel):
+    """LlamaModel with a paged KV cache and supporting variable sequence
+    length batched inference.
 
-class DirectCacheLlamaModelV1(ThetaLayer):
```

Comment:
Any thoughts on leaving the naive implementation on for reference/education? :)

Reply:
It's still there in the sibling file for that reason... Just wanted to have less to look at in this file

Reply:
nice!